### PR TITLE
refactor: more constants, improved types

### PR DESCRIPTION
**Pull request by bvandercar-vt**
- remove many `any` types by actually specifying the correct type
- add more constants to `CONSTANTS`
- improve link between `CONSTANTS` and types

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,13 +1,58 @@
...
+
+    state(key: 'done', callback: (error: Error) => Promise<void>): void
+    state(key: 'error', error: Error): void
+  }
+
+  interface Chainable<Subject = any> extends Pick<Promise<Subject>, 'catch'> {
+  }
+}
+
+declare namespace Mocha {
```
Comment (archfz) [inline index.d.ts:31]: I think we should do this differently. These types are not part of the public api of cypress-terminal-report. This file should be just for the public global types.

You can however create another d.ts that we can include specifically in tsconfig for types, to be used only internally.

Comment (bvandercar-vt) [reply index.d.ts]: That's a great point, I didn't realize this file is included in the package. Moved these new ones to a separate file that I added to `.npmignore`: https://github.com/archfz/cypress-terminal-report/pull/261/commits/bf84798fc0245fa0e58d57fecd992cc2429ab1ee

**Comment by archfz** [comment]
This looks really good. Nice job.

**Comment by archfz** [comment]
Should this be merged before the lint update? Thinking of conflicts.

**Pull request merged**